### PR TITLE
Move instance of http client to backend instance

### PR DESCRIPTION
**Pull request by mnezerka**
Keep one single instance of http client and reuse
it for all outgoing http requests. This change was
initiated by observation of resources leek: nubmer
of "closed" tcp socket file descriptors were incremented
by each set of auth requests (user, supervisor, acl)

This PR is related to #22

### CODE REVIEW

File: backends/http.go [hunk 1]
```diff
@@ -37,12 +39,21 @@ func NewHTTP(authOpts map[string]string, logLevel log.Level) (HTTP, error) {
 
 	log.SetLevel(logLevel)
 
+	tr := &h.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
```
Comment (iegomez) [inline backends/http.go]: Whether peer verification is to be skipped should depend on the verifyPeer option.

File: backends/http.go [hunk 2]
```diff
@@ -158,11 +169,11 @@ func (o HTTP) CheckAcl(username, topic, clientid string, acc int32) bool {
 		"acc":      []string{strconv.Itoa(int(acc))},
 	}
 
-	return httpRequest(o.Host, o.AclUri, username, o.WithTLS, o.VerifyPeer, dataMap, o.Port, o.ParamsMode, o.ResponseMode, urlValues)
+	return o.httpRequest(o.Host, o.AclUri, username, o.WithTLS, o.VerifyPeer, dataMap, o.Port, o.ParamsMode, o.ResponseMode, urlValues)
 
 }
 
-func httpRequest(host, uri, username string, withTLS, verifyPeer bool, dataMap map[string]interface{}, port, paramsMode, responseMode string, urlValues map[string][]string) bool {
+func (o HTTP) httpRequest(host, uri, username string, withTLS, verifyPeer bool, dataMap map[string]interface{}, port, paramsMode, responseMode string, urlValues map[string][]string) bool {
```
Comment (iegomez) [inline backends/http.go]: This function no longer needs params that are associated to the HTTP backend as it's now a method of it and it can access them.

**Comment by mnezerka** [comment]
@iegomez Thank you for review. I tried to address all your points in amended commit dfc14cf.

**Comment by iegomez** [comment]
@mnezerka Thank you for taking the time.

**Pull request merged**